src/app/game/header/Score.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useRef } from 'react';
+
+import { timeFormat } from '../../helpers';
+import { useActions, useTypedSelector } from '../../hooks';
+import { icons } from '../../icons';
+import { Styled } from './styled';
+
+const msForUpdate = 1000;
+
+export const Score = () => {
+  const [time, isPaused] = useTypedSelector((store) => [
+    store.user.time,
+    store.game.isPaused,
+  ]);
+  const { updateTime } = useActions();
+  const increment = useRef<NodeJS.Timeout | null>(null);
+
+  const handleStart = () => {
+    increment.current = setInterval(() => {
+      updateTime(time + 1);
+    }, msForUpdate);
+  };
+
+  const handleStop = () => {
+    if (increment.current) {
+      clearInterval(increment.current);
+    }
+  };
+
+  useEffect(() => {
+    if (!isPaused) {
+      handleStart();
+    }
+
+    return () => {
+      handleStop();
+    };
+  });
```

Comment:
Do we need to call useEffect on every change? Or maybe we could add some args?
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

For example `[isPaused]`?